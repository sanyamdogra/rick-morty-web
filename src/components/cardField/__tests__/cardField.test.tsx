import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import CardField from "..";

afterEach(cleanup);

test("should match the snapshot and render correct data", () => {
  const { asFragment } = render(
    <CardField label="test-label" value="test-value" />
  );

  expect(asFragment()).toMatchSnapshot();

  expect(screen.getByTestId("card-field-label")).toHaveTextContent(
    "test-label"
  );

  expect(screen.getByTestId("card-field-value")).toHaveTextContent(
    "test-value"
  );
});
