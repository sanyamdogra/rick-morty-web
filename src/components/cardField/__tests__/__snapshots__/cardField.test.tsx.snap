// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot and render correct data 1`] = `
<DocumentFragment>
  <div
    class="rick-morty-card-field__label"
    data-testid="card-field-label"
  >
    test-label
  </div>
  <div
    class="rick-morty-card-field__value rick-morty-card-field__value--none"
    data-testid="card-field-value"
  >
    test-value
  </div>
</DocumentFragment>
`;
